internal/controllers/addon/phase_observe_csv.go

```diff
@@ -19,6 +19,11 @@ func (r *olmReconciler) observeCurrentCSV(
 	if err := r.uncachedClient.Get(ctx, csvKey, csv); err != nil {
 		return resultNil, fmt.Errorf("getting installed CSV: %w", err)
 	}
+	// If the addon was being upgraded, we mark the upgrade as
+	// concluded.
+	if addonUpgradeConcluded(addon, csv) {
+		reportAddonUpgradeSucceeded(addon)
+	}
```

Comment:
i am wondering whether this check should be done only if csv has succeeded.

Reply:
addonUpgradeConcluded func already has that check